src/libraries/System.Private.CoreLib/src/System/Globalization/OrdinalCasing.Icu.cs

```diff
@@ -437,7 +462,7 @@ internal static unsafe int LastIndexOf(ReadOnlySpan<char> source, ReadOnlySpan<c
                 char* pTable = (char*)table;
                 Interop.Globalization.InitOrdinalCasingPage(pageNumber, pTable);
             }
-            s_casingTable[pageNumber] = casingTable;
+            s_casingTable![pageNumber] = casingTable;
```

Comment:
Existing bug: this really should be `Volatile.Write(ref s_casingTable[pageNumber], casingTable)`. We want the page's table to be published only after the table's contents have been populated.

Technically there's currently no official runtime that works on a weak memory model, but may as well stave off possible future problems while we're here.